### PR TITLE
Gh 4581 remove jerven design mistakes

### LINKED ISSUE #4581: Remove my own design errors with v5
### Current Behavior

There are a few errors in design that are already deprecated but where originally introduced my myself. With the wisdom of a few years more I would like to remove them.
Chief among them are the LimitedSize strategy, and makeSet methods on the EvaluationStrategy interface.

### Expected Behavior

Remove the deprecated code.

### Steps To Reproduce

_No response_

### Version

4.x

### Are you interested in contributing a solution yourself?

Yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by JervenBolleman**
GitHub issue resolved: # 

Briefly describe the changes proposed in this PR:

Remove LimitedSize evaluation strategy code, as well as a few methods in the EvaluationStrategy API that was replaced by the collection factory API.

Closes #4581

### CODE REVIEW

**Comment by JervenBolleman** [comment]
@hmottestad could this still be included in the next 5.x Milestone build?

**Comment by hmottestad** [comment]
> @hmottestad could this still be included in the next 5.x Milestone build?

Yeah. I haven't started on the build process yet. I saw you requested a review from me, so I'll take a look asap.

File: core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/DefaultEvaluationStrategy.java [hunk 1]
```diff
@@ -710,10 +709,26 @@ protected QueryEvaluationStep prepare(DescribeOperator node, QueryEvaluationCont
...
+				final CloseableIteration<BindingSet> evaluate = child.evaluate(bindings);
+				return new DistinctIteration<BindingSet>(evaluate,
+						cf.createSetOfBindingSets()) {
+
+					@Override
+					protected void handleClose() throws QueryEvaluationException {
+						try {
+							cf.close();
+						} catch (QueryEvaluationException e) {
+							super.handleClose();
```
Comment (hmottestad) [inline core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/DefaultEvaluationStrategy.java:726]: Should this maybe be try-finally or is super.handleClose() only supposed to be called when there is an exception?

Comment (JervenBolleman) [reply core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/DefaultEvaluationStrategy.java:726]: This should be a try-finally, although right now super.handleClose is a noop. That could change in the future.
will-fix.

**Comment by hmottestad** [comment]
From what I understand looking through the code you've moved the set creation to a higher level in the code (EvaluationFactory?) so that it's easier to override in the specific stores. 

Only issue I found in your code was the try-catch instead of try-finally. If you're happy with everything then you can go ahead and merge.

**Comment by JervenBolleman** [comment]
> From what I understand looking through the code you've moved the set creation to a higher level in the code (EvaluationFactory?) so that it's easier to override in the specific stores.
> 
> Only issue I found in your code was the try-catch instead of try-finally. If you're happy with everything then you can go ahead and merge.

Yes. Exactly, now with collection factories one does not need to extend the store. One can just inject the limited size collections. Also now there are fewer and fewer places left where collections are not coming from a collection factory that can spill to disk. So the original goal I had with these LimitedSize* all those years ago is no longer valid.

**Pull request merged**